edgelesssys/constellation
# terraform: azure node groups

### Context

This is the first step towards supporting multiple groups of control plane and worker nodes (with different instance configuration and zones).
Similar to #1941.

Additionally, we found a bug in the current terraform migration behavior that allows any target when the target slice is empty.

### Proposed change(s)

- terraform: azure node groups
- add a check to skip terraform migration when the targets are empty



### How to test
as in #1941:

Check that an existing azure cluster on v2.8 can be migrated to this terraform state without any resources being deleted.
Manually add an additional node group with different settings (zone, instance type) to the terraform.tfvars file. It should be created successfully using terraform apply (although it can't join the Constellation yet).

Correct creation and deletion using the PR-CLI was also verified.

### Checklist
